Add sponsored search support to search page and site-navbar

B77Mills opened the pull request:
- Add support for setting a searchSponsor within the site config.
 ```js
 sponsorLogos: {
  navbar: {
    alt: 'Search Sposored by ProfiNet',
    src: 'https://img.automationworld.com/files/base/pmmi/aw/profinet_black.png?h=24&auto=format,compress',
    srcset: [
      'https://img.automationworld.com/files/base/pmmi/aw/profinet_black.png?h=24&auto=format,compress&dpr=2 2x',
    ],
  },
  page: {
    alt: 'Search Sposored by ProfiNet',
    src: 'https://img.automationworld.com/files/base/pmmi/aw/profinet_green.png?h=48&auto=format,compress',
    srcset: [
      'https://img.automationworld.com/files/base/pmmi/aw/profinet_green.png?h=48&auto=format,compress&dpr=2 2x',
    ],
  },
},
```
- Add support to the site-navbar call to display search icon & conditionally display the sponsor logo when set.  
- Add ability to also tell the site-menu-toggle to display left or right
- Example site call looks like `<theme-site-header has-user=hasUser reg-enabled=isEnabled show-search-icon=true menu-btn-position="right"/>`
<img width="1205" alt="Screen Shot 2022-06-30 at 8 59 41 AM" src="https://user-images.githubusercontent.com/3845869/176696676-08baa458-96d8-4fa5-8dc9-38eb3b15687f.png">

B77Mills committed:
add search route to example website

B77Mills committed:
add site-navbar search component

New component will display the search icon and link to /search by default.  You can also configure a sponsor(logo) to display to the right of the icon.

B77Mills committed:
Update default monorail site-header with icon-wrapper

Add new icon-wrapper element the now holds the icons that appear on the right.  At the moment it will contain search, newsletterToggler & site-menu-toggle(when positioned right).  I also added support for passing a menuBtnPosition prop left || right.  If left it display like it use to.  If right it will pull the menu toggler into the icon-wrapper as the last item.

alls added this setup to example site

B77Mills committed:
Add searchSponsor support to search page template

If sponsored it will pull the title into its own row with the `Sponsored by LOGO`  block to the right

B77Mills committed:
Add Support for both site-menu and site-navebar

Add support to display the sponor logo both in the site-navbar & in the site-menu components.

B77Mills committed:
Moved this inside the if check

brandonbk commented on packages/marko-web-theme-monorail/components/site-menu/components/search.marko:
```diff
@@ -12,6 +14,14 @@ $ const inputId = randomElementId({ prefix: "site-menu-search" });
+          alt=sponsor.siteMenuLogo.alt
+          src=sponsor.siteMenuLogo.src
+          srcset=sponsor.siteMenuLogo.srcset
+          class="search-sponsor__logo lazyload"
```
Remove lazyload from this since it is above the fold

B77Mills committed:
update wehre searchSponsor config lives

Move it to site.search.sponsorLogos from site.searhSponsors

B77Mills committed:
adjust sponsored search navbar/menu styles

B77Mills committed:
adjust the order of the navbar icons.

make it newsletter || search || menu when configured on the right side of the header

B77Mills committed:
add padding to left side of icon wrapper

B77Mills commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add Search to top nav, remove from hamburger nav and we need sponsored search for AW](https://trello.com/c/dWBy0hVn/353-add-search-to-top-nav-remove-from-hamburger-nav-and-we-need-sponsored-search-for-aw)

B77Mills committed:
search page cleanup

remove console.log
add ad calls back in.
